packages/near-fast-auth-signer/src/components/AuthCallback/AuthCallback.tsx

```diff
@@ -1,290 +1,164 @@
+          await window.fastAuthController.setKey(keyPair);
+        }
 
-          await window.fastAuthController.claimOidcToken(accessToken);
-          const oidcKeypair = await window.fastAuthController.getKey(`oidc_keypair_${accessToken}`);
-          if (!window.firestoreController) {
-            window.firestoreController = new FirestoreController();
-          }
-          window.firestoreController.updateUser({
-            userUid:   user.uid,
-            oidcToken: accessToken,
-          });
+        if (!window.fastAuthController.getAccountId()) {
+          window.fastAuthController.setAccountId(params.accountId);
+        }
 
-          const callback = isRecovery ? onSignIn : onCreateAccount;
-          await callback({
-            oidcKeypair,
-            accessToken,
-            accountId,
-            publicKeyFak,
-            public_key_lak,
-            contract_id,
-            methodNames,
-            success_url,
-            setStatusMessage,
-            navigate,
-            searchParams,
-            gateway:      success_url,
-          });
-        } catch (e) {
-          captureException(e);
-          setCallbackError(e);
+        await window.fastAuthController.claimOidcToken(accessToken);
+        const oidcKeypair = await window.fastAuthController.getKey(`oidc_keypair_${accessToken}`);
+        if (!window.firestoreController) {
+          window.firestoreController = new FirestoreController();
         }
-      } else {
-        navigate('/signup');
+        window.firestoreController.updateUser({
+          userUid:   user.uid,
+          oidcToken: accessToken,
+        });
+
+        const callback = params.isRecovery ? onSignIn : onCreateAccount;
+        await callback({
+          oidcKeypair,
+          accessToken,
+          accountId:         params.accountId,
+          publicKeyFak,
+          public_key_lak:    params.public_key_lak,
+          contract_id:       params.contract_id,
+          methodNames:       params.methodNames,
+          success_url:       params.success_url,
+          setStatusMessage,
+          navigate,
+          searchParams,
+          gateway:           params.success_url,
+          onAccountNotFound: () => {
+            setAccountAsExisting(false);
+          },
+        });
+      } catch (e) {
+        captureException(e);
+        setCallbackError(e);
       }
     };
 
     signInProcess();
   }, [navigate, searchParams]);
 
-  if (callbackError) return <AuthCallbackError error={callbackError} redirectUrl={decodeIfTruthy(searchParams.get('failure_url'))} />;
-
-  return <StyledStatusMessage data-test-id="callback-status-message">{statusMessage}</StyledStatusMessage>;
+  if (!isAccountExisting) {
+    return (
+      <PageWrap>
+        <AccountInfo>Oops! This account doesn&apos;t seem to exist. Please create one below.</AccountInfo>
+        <CreateAccountForm
+          onSubmit={createAccount}
```

Comment:
Yes @benmalcom we already have the firebase token in this scope so no need to resend the email again we just need to prompt the user to choose an accountId and talk to mpc to create it